### PR TITLE
Fixed problem with extensions being always loaded with default options

### LINKED ISSUE #1716: Extensions are loaded with default options
In loaded extension there is no access to config by options in console. 

For example:

    $this->output->getVerbosity()

always return '1' even if script is started with '-vvv'

As I found in https://github.com/Codeception/Codeception/blob/master/src/Codeception/Codecept.php#L65 extensions are loaded before merging config from file and constructor parameter.

### PR DESCRIPTION
**Pull request by sjableka**
Fixes #1716

### CODE REVIEW

**Comment by DavertMik** [comment]
This break current test suite. Please run `codecept run cli` to check how the test differs from your PR

**Comment by sjableka** [comment]
@DavertMik Sorry about that, it is fine now.

File: src/Codeception/Codecept.php [hunk 1]
```diff
@@ -57,12 +57,14 @@ class Codecept
...
+
+        $baseOptions = $this->mergeOptions($options);
+
+        $this->loadExtensions($baseOptions);
 
         $this->config = Configuration::config();
 
-        $this->options = array_merge($this->options, $this->config['settings']);
-        $this->options = array_merge($this->options, $options);
+        $this->options = $this->mergeOptions($options);
```
Comment (DavertMik) [inline src/Codeception/Codecept.php]: is there a reason why `mergeOptions` is executed 2 times on the same data?
can merged options be reused?

**Comment by sjableka** [comment]
It's not the same data. It calls Configuration::config() to get current configuration, which may be changed during extenstion loading. Without calling it twice, tests using SimpleOutput class are failing.

File: src/Codeception/Codecept.php (hunk 1 above)
Comment (sjableka) [inline src/Codeception/Codecept.php]: @DavertMik you can find my answer below ;)

**Comment by DavertMik** [comment]
Ok, got it. Thanks.

**Pull request merged**